test/hash.cxx

```diff
@@ -7,6 +7,52 @@
 #include <substrate/hash>
 #include <catch2/catch_test_macros.hpp>
 
+TEST_CASE("murmur128", "[hash]")
+{
+	const std::string empty{};
+	const std::array<uint8_t, 1> zero{{0x0}};
+	const std::string example{"The quick brown fox jumps over the lazy dog"};
+	const std::string example2{"The quick brown fox jumps over the lazy dog."};
 
+	const auto pair1{substrate::murmur128(empty, 0x900DBEEF)};
+	const auto pair2{substrate::murmur128(zero, 0x900DBEEF)};
+	const auto pair3{substrate::murmur128(example, 0x900DBEEF)};
+	const auto pair4{substrate::murmur128(example2, 0x900DBEEF)};
+
+	REQUIRE(pair1.first == 0xd80062720a4a93eb);
```

Comment:
Given these are 64-bit numbers, and to prevent issues (or warnings) on some platforms, it might be worth wrapping them in `UINT64_C()`

Reply:
Ack, btw, do you have any specifics on the casing of the constants?

Reply:
The way that [integer literals are defined to get typed](https://en.cppreference.com/w/cpp/language/integer_literal#The_type_of_the_literal) means that unprefixed literals can wind up unable to be represented by a wide enough type in a platform-dependant manner, and even if that doesn't happen, might be a different type to the `.first` member.

Simply wrapping as in `UINT64_C(0xd80062720a4a93eb)` should be sufficient to prevent all this and the associated warnings (albeit for the warnings to appear post-C++11 means the implementation's not fully conforming)